core/capabilities/ccip/ccipevm/addresscodec.go

```diff
@@ -1,23 +1,20 @@
 package ccipevm
 
 import (
-	"encoding/hex"
 	"fmt"
-	"strings"
+
+	"github.com/ethereum/go-ethereum/common"
 )
 
 type AddressCodec struct{}
 
 func (a AddressCodec) AddressBytesToString(addr []byte) (string, error) {
-	// TODO support EIP-55 checksum, https://smartcontract-it.atlassian.net/browse/CCIP-5340
-	return "0x" + hex.EncodeToString(addr), nil
+	return common.BytesToAddress(addr).Hex(), nil
```

Comment:
So what was the root issue here?

Reply:
Pretty sure this was just because we can. We would have implemented it like this in chainlink-ccip if we had access to go-ethereum.

Reply:
Didn't this break things before? That's why we had the ticket right?

Reply:
Yeah it required the change in deployments tooling so that the onchain state is configured correctly. Since I fixed that at the same time it's no longer broken, but it does mean that you'll have to update the onchain DON config on your staging deployment